relay/sources/rudder-package/src/lib.rs

```diff
@@ -81,19 +83,47 @@ pub fn run() -> Result<()> {
 }
 
 pub mod action {
-    use anyhow::{bail, Result};
+    use anyhow::{anyhow, bail, Result};
+    use log::debug;
 
     use crate::archive::Rpkg;
     use crate::database::Database;
+    use crate::repo_index::RepoIndex;
+    use crate::repository::Repository;
+    use crate::versions::RudderVersion;
     use crate::webapp_xml::restart_webapp;
-    use crate::PACKAGES_DATABASE_PATH;
+    use crate::{
+        PACKAGES_DATABASE_PATH, REPOSITORY_INDEX_PATH, RUDDER_VERSION_PATH, TMP_PLUGINS_FOLDER,
+    };
     use std::path::Path;
 
-    pub fn install(force: bool, package: String) -> Result<()> {
+    pub fn install(force: bool, package: String, repository: Repository) -> Result<()> {
         let rpkg_path = if Path::new(&package).exists() {
             package
         } else {
-            bail!("TODO");
```

Comment:
just for information, there is a built-in `todo!()` macro for this purpose.

Reply:
:+1: